internal/artifactref/artifactref.go

```diff
@@ -0,0 +1,11 @@
+package artifactref
+
+import "regexp"
+
+// case 1: steps.<step-name>.inputs
+// case 2: steps.<step-name>.outputs
+// case 3: steps.<step-name>.inputs.<artifact-category-name>
+// case 4: steps.<step-name>.outputs.<artifact-category-name>
+const stepArtifactUsagePattern = `\$\(steps\.([^.]+)\.(?:inputs|outputs)(?:\.([^.]+))?\)`
```

Comment:
There is an issue with the second `[^.]+` - the regex is too eager and in case of a second `)` at the end, the group eats away the first `)`.

I tried the following consumer step:

```yaml
      - name: artifacts-consumer-python
        image: python:latest
        script: |
          #!/usr/bin/env python3
          import json

          data = json.loads($(steps.artifacts-producer.outputs.image))
          print(f'URL: {data["url"]}, Digest: {data["digest"]}')
```

and it fails with:
```
"msg":"Error while substituting step artifacts: values for template $(steps.artifacts-producer.outputs.image))
```

Note the double `)` at the end.
A working regex could be:

```
\$\(steps\.([^.]+)\.(?:inputs|outputs)(?:\.([^.^\)]+))?\)
```

Reply:
I can change the suggested regex, but you need to quote $(steps.artifacts-producer.outputs.image) this '$(steps.artifacts-producer.outputs.image)' in your example